packages/block-editor/src/components/block-list/use-block-props/use-event-handlers.js

```diff
@@ -108,6 +108,10 @@ export function useEventHandlers( clientId ) {
 			}
 
 			function onMouseLeave( { buttons } ) {
+				if ( ! isBlockSelected( clientId ) ) {
+					return;
```

Comment:
I'm not sure if it's a good idea to add this event handler to every block? It's only necessary for the selected block, there's no need to track mouse leaving for all the other blocks. Maybe there's barely a difference in performance though.